src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -34,14 +34,21 @@ class DelegatingLoader extends BaseDelegatingLoader
     /**
      * Constructor.
      *
+     * Ability to pass a LoggerInterface instance as second argument will be removed in 3.0.
+     *
      * @param ControllerNameParser    $parser   A ControllerNameParser instance
-     * @param LoggerInterface         $logger   A LoggerInterface instance
      * @param LoaderResolverInterface $resolver A LoaderResolverInterface instance
      */
-    public function __construct(ControllerNameParser $parser, LoggerInterface $logger = null, LoaderResolverInterface $resolver)
+    public function __construct(ControllerNameParser $parser, $resolver, $r = null)
     {
         $this->parser = $parser;
-        $this->logger = $logger;
+
+        if ($resolver instanceof LoggerInterface) {
```

Comment:
this is not BC in case ``$resolver`` is ``null`` and the resolver is passed in ``$r`` (i.e. using the old API without the logger). IMO, the code should be like this:

```php
if (!$resolver instanceof LoaderResolverInterface) {
    $this->logger = $resolver;
    $resolver = $r;
}
```

And keeping the typehint on the ``$r`` argument (while making it nullable), to ensure that you don't pass other weird thing there